Src/LinearSolvers/MLMG/AMReX_MLTensorOp_grad.cpp

```diff
@@ -95,33 +140,36 @@ MLTensorOp::compVelGrad (int amrlev, const Array<MultiFab*,AMREX_SPACEDIM>& flux
 
     const int mglev = 0;
 
-    applyBCTensor(amrlev, mglev, sol, BCMode::Inhomogeneous, StateMode::Solution, m_bndry_sol[amrlev].get());
+    MLMGBndry const* bndry = m_bndry_sol[amrlev].get();
+    applyBC(amrlev, mglev, sol, BCMode::Inhomogeneous, StateMode::Solution, bndry);
+    applyBCTensor(amrlev, mglev, sol, BCMode::Inhomogeneous, StateMode::Solution, bndry);
+
+    const auto& bcondloc = *m_bcondloc[amrlev][mglev];
 
     const auto dxinv = m_geom[amrlev][mglev].InvCellSizeArray();
-    const int dim_fluxes = AMREX_SPACEDIM*AMREX_SPACEDIM;
+    const Box& domain = m_geom[amrlev][mglev].growPeriodicDomain(1);
+    const auto dlo = amrex::lbound(domain);
+    const auto dhi = amrex::ubound(domain);
 
 #ifdef AMREX_USE_OMP
 #pragma omp parallel if (Gpu::notInLaunchRegion())
 #endif
+    for (MFIter mfi(sol, TilingIfNotGPU()); mfi.isValid(); ++mfi)
     {
-        Array<FArrayBox,AMREX_SPACEDIM> fluxfab_tmp;
+        Array4<Real const> const vfab = sol.const_array(mfi);
+        AMREX_D_TERM(Box const xbx = mfi.nodaltilebox(0);,
+                     Box const ybx = mfi.nodaltilebox(1);,
+                     Box const zbx = mfi.nodaltilebox(2);)
+        AMREX_D_TERM(Array4<Real> const fxfab = fluxes[0]->array(mfi);,
+                     Array4<Real> const fyfab = fluxes[1]->array(mfi);,
+                     Array4<Real> const fzfab = fluxes[1]->array(mfi);)
```

Comment:
This should be `Array4<Real> const fzfab = fluxes[2]->array(mfi);` right

Reply:
Yes.